packages/client/src/DataCommonsClient.ts

```diff
@@ -0,0 +1,480 @@
+  }
+
+  async getFirstNodeValues(params: {
+    dcids: string[];
+    prop: string;
+  }): Promise<Record<string, string | null>> {
+    const nodePropvals = await this.getNodePropvals(params);
+    const nodeValues: Record<string, string | null> = {};
+    Object.keys(nodePropvals).forEach((variableDcid) => {
+      nodeValues[variableDcid] =
+        nodePropvals[variableDcid].length > 0
+          ? nodePropvals[variableDcid][0].value
+          : null;
+    });
+    return nodeValues;
+  }
+
+  async getNodePropvals(params: {
+    dcids: string[];
+    prop: string;
+  }): Promise<ApiNodePropvalOutResponse> {
+    const queryString = toURLSearchParams({
+      dcids: params.dcids,
+      prop: params.prop,
+    });
+    const url = `${this.apiRoot || ""}/api/node/propvals/out?${queryString}`;
+    return (await (await fetch(url)).json()) as ApiNodePropvalOutResponse;
+  }
+
+  /**
+   * Fetches place/entity names
+   * @param entityDcids list of entity DCIDs
+   * @param prop optional entity name property
+   */
+  async getPlaceNames(params: {
+    dcids: string[];
+    prop?: string;
+  }): Promise<Record<string, string>> {
+    if (!params.dcids.length) {
+      return Promise.resolve({});
+    }
+    const url = `${this.apiRoot || ""}/api/place/name`;
+    const response = await fetch(url, {
+      method: "post",
+      body: JSON.stringify({
+        dcids: params.dcids,
+        prop: params.prop,
+      }),
+      headers: {
+        "Content-Type": "application/json",
+      },
+    });
+    return (await response.json()) as Record<string, string>;
+  }
+
+  /**
+   * Gets and processes the data from /api/observations/point endpoint
+   * @param apiRoot api root
+   * @param entities list of entitites to get data for
+   * @param variables list of variables to get data for
+   * @param date date to get the data for
+   * @param alignedVariables groups of variables that should have the same unit
+   */
+  async getObservationsPoint(params: {
+    date?: string;
+    entities: string[];
+    variables: string[];
+  }): Promise<PointApiResponse> {
+    const queryString = toURLSearchParams({
+      date: params.date,
+      entities: params.entities,
+      variables: params.variables,
+    });
+    const url = `${this.apiRoot || ""}/api/observations/point?${queryString}`;
+    return (await (await fetch(url)).json()) as PointApiResponse;
+  }
+
+  /**
+   * Fetches point observations related to entities of type childType
+   * that are contained within the specified parentEntity
+   *
+   * Uses /api/observations/point/within endpoint
+   * @param childType type of entity to get data for
+   * @param parentEntity the parent entity of the entities to get data for
+   * @param variables list of variables to get data for
+   * @param date date to get the data for
+   */
+  async getObservationsPointWithin(params: {
+    parentEntity: string;
+    childType: string;
+    variables: string[];
+    date?: string;
+    facetIds?: string[];
```

Comment:
include this in the comment doc list of params?

Reply:
Removed facetIds entirely for now (revisiting in a future PR)